src/backend/main/java/com/example/boilerscout/api/ProfileController.java

```diff
@@ -109,4 +109,109 @@
     }
 
 
+    public Map<String, Object> getProfile(String userId, String token, String goal) {
```

Comment:
Nitpick: can you rename `goal` to like `query` or something? I assume `goal` is just the userId of the person you want to look at